gimp.lua

```diff
@@ -0,0 +1,186 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+--[[
+    gimp - export an image and open with GIMP for editing
+
+    This script provides another storage (export target) for darktable.  Selected
+    images are exported in the specified format to temporary storage.  GIMP is launched
+    and opens the files.  After editing, the exported images are overwritten to save the 
+    changes.  When GIMP exits, the exported files are moved into the current collection
+    and imported into the database.  The imported files then show up grouped with the 
+    originally selected images.
+
+    ADDITIONAL SOFTWARE NEEDED FOR THIS SCRIPT
+    * GIMP - http://www.gimp.org
+
+    USAGE
+    * require this script from your main lua file
+    * select an image or images for editing with GIMP
+    * in the export dialog select "Edit with GIMP" and select the format and bit depth for the
+      exported image
+    * Press "export"
+    * Edit the image with GIMP then save the changes with File->Overwrite....
+    * Exit GIMP
+    * The edited image will be imported and grouped with the original image
+
+    CAVEATS
+    * Developed and tested on Ubuntu 14.04 LTS with darktable 2.0.3 and GIMP 2.9.3 (development version with
+      > 8 bit color)
+    * There is no provision for dealing with the xcf files generated by GIMP, since darktable doesn't deal with 
+      them.  You may want to save the xcf file if you intend on doing further edits to the image or need to save 
+      the layers used.  Where you save them is up to you.
+
+    BUGS, COMMENTS, SUGGESTIONS
+    * Send to Bill Ferguson, wpferguson@gmail.com
+]]
+
+local dt = require "darktable"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("gimp",dt.configuration.config_dir.."/lua/")
+
+-- Thanks to http://lua-users.org/wiki/SplitJoin for the split and split_path functions
+local function split(str, pat)
+   local t = {}  -- NOTE: use {n = 0} in Lua-5.0
+   local fpat = "(.-)" .. pat
+   local last_end = 1
+   local s, e, cap = str:find(fpat, 1)
+   while s do
+      if s ~= 1 or cap ~= "" then
+        table.insert(t,cap)
+      end
+      last_end = e+1
+      s, e, cap = str:find(fpat, last_end)
+   end
+   if last_end <= #str then
+      cap = str:sub(last_end)
+      table.insert(t, cap)
+   end
+   return t
+end
+
+local function split_path(str)
+   return split(str,'[\\/]+')
+end
+
+local function get_filename(str)
+  parts = split_path(str)
+  return parts[#parts]
+end
+
+local function basename(str)
+  return string.sub(str,1,-4)
```

Comment:
That is an awesome regular expression.  I had to fire up lua in a shell and play with it for awhile to figure it out.  Well done, and thanks.